src/cartesian/ReferenceLine.tsx

```diff
@@ -192,15 +192,23 @@ export function ReferenceLine(props: Props) {
   );
 }
 
-ReferenceLine.displayName = 'ReferenceLine';
-ReferenceLine.defaultProps = {
-  isFront: false,
-  ifOverflow: 'discard',
-  xAxisId: 0,
-  yAxisId: 0,
-  fill: 'none',
-  stroke: '#ccc',
-  fillOpacity: 1,
-  strokeWidth: 1,
-  position: 'middle',
-};
+// eslint-disable-next-line react/prefer-stateless-function -- requires static defaultProps
+export class ReferenceLine extends React.Component {
```

Comment:
You need to reference Props :) 
```
export class ReferenceLine extends React.Component<Props> {
```

Reply:
Yeah I should've caught that!

Reply:
I can fix this when I push some other things

Reply:
Is this what's causing

```
Property 'x' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<ReferenceLine> & Pick<Readonly<{}>, never> & InexactPartial<...> & InexactPartial<...>'
```
?

Reply:
@oscar-b That's correct. https://github.com/recharts/recharts/pull/4542#issuecomment-2138356968

Reply:
Resolved in latest alpha